crates/supervisor/core/src/l1_watcher/watcher.rs

```diff
@@ -138,32 +143,65 @@ where
         }
     }
 
-    fn handle_new_latest_block(&self, incoming_block: Block, previous_block: &mut BlockNumHash) {
+    async fn handle_new_latest_block(
+        &self,
+        incoming_block: Block,
+        previous_block: &mut BlockNumHash,
+    ) {
         let incoming_block_number = incoming_block.header.number;
+
+        // Early exit if the incoming block is not newer than the previous block
         if incoming_block_number <= previous_block.number {
             info!(
                 target: "l1_watcher",
+                incoming_block_number,
+                previous_block_number = previous_block.number,
                 "Incoming latest L1 block is not greater than the stored latest block"
             );
             return;
         }
 
+        info!(
```

Comment:
```suggestion
        trace!(
```
this granularity is better suited for trace log level